_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,505 @@
+
+To create a `Setup()` function, we just define this function in our fixture class which will [override](https://en.cppreference.com/w/cpp/language/override) the [virtual function](https://en.cppreference.com/w/cpp/language/virtual) in `testing::Test` class in GoogleTest.
+
+For our table class, we can create the `Setup()` and `Teardown()` functions as shown below. For more details, please see [4_table_test_with_setup.cpp](../code/Chapter3/4_table_test_with_setup.cpp).
+
+```cpp
+// Test fixture for EmployeeTable class.
+class EmployeeTableTest : public testing::Test {
+    public:
+        EmployeeTable table;
+};
+
+TEST_F(EmployeeTableTest, TableIsEmptyWhenCreated) {
+    EXPECT_TRUE(table.isEmpty());
+}
+
+TEST_F(EmployeeTableTest, TableHasSizeZeroWhenCreated) {
+    EXPECT_EQ(table.getEntryCount(), 0);
+}
+
+class EmployeeTableWithOneEmployee : public testing::Test {
+    public:
+        EmployeeTable table;
+        Employee* employee;
+
+        void SetUp() override {
+            employee = new Employee("John Doe", 30, 5000, 5, 1000);
+            table.addEmployee(*employee);
+        }
+
+        void TearDown() override {
+            delete employee;
+            employee = nullptr;
+        }
+};
+
+TEST_F(EmployeeTableWithOneEmployee, TableIsNotEmptyWhenCreatedWithOneEmployee) {
+    EXPECT_FALSE(table.isEmpty());
+}
+
+TEST_F(EmployeeTableWithOneEmployee, NumberOfEntriesIsOneWhenCreatedWithOneEmployee) {
+    EXPECT_EQ(table.getEntryCount(), 1);
+}
+
+TEST_F(EmployeeTableWithOneEmployee, NumberOfEntriesIsOneLessAfterRemovingEmployee) {
+    table.removeEmployee("John Doe");
+    EXPECT_EQ(table.getEntryCount(), 0);
+}
+```
+
+As we can see from above, our test looks much cleaner with the setup and teardown function. The reason for creating another test fixture for writing `Setup()` and `Teardown()` is because the first two tests do not require it.
+
+## 7. Tests Filtering in GoogleTest
+
+Sometimes, we may have some tests that take a lot of time to run. In some other case, when we are developing and testing our code, we do not want to run our entire test suite and run only one of the test that we have recently added.
+
+GoogleTest allows to filter test by command line parameter `--gtest_filter`. The general syntax to use `gtest_filter` is
+
+```bash
+$ ./your_executable --gtest_filter=Pattern[-Pattern]
+```
+
+where `Pattern` can be a valid string patterns. The `-Pattern` will run all tests except the pattern in the command. Instead of a pattern, we can also use full test name in the form `test_suite_name.test_name`.
+
+Let us run our table tests in the file [4_table_test_with_setup.cpp](../code/Chapter3/4_table_test_with_setup.cpp). Let us assume that the executable name is `my_test`. We get the following output.
```

Comment:
This section is useful, but I wonder if it should be separated into its own section, or maybe put into another section where you introduce running gTest binaries?

Reply:
It is a small subsection and I would like to keep it here for now.